Added tests

gestia535 opened the pull request:
Added file with fixtures - conftest.py
Added files with data and helpers.py
Added file README.md
Added allure report and requirements.txt

gestia535 committed:
Added tests
Added file with fixtures - conftest.py
Added files with data and helpers.py
Added file README.md
Added allure report and requirements.txt

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,43 @@
+
+@pytest.fixture()
+def create_and_login_user():
+    payload = {'email': login_generator(), 'password': password_generator(), 'name': name_generator()}
+    requests.post(CREATE_USER, json=payload)
```
Нужно исправить: эта часть кода как и удаление - это копипаста фикстуры выше, необходимо вызывать ее и делать только логин а не копипастить

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,43 @@
+
+@pytest.fixture()
+def create_login_and_order():
+    payload = {'email': login_generator(), 'password': password_generator(), 'name': name_generator()}
+    requests.post(CREATE_USER, json=payload)
+    login_response = requests.post(LOGIN_USER, json=payload)
+    access_token = login_response.json().get('accessToken')
+    order_payload = correct_ingredients
```
up

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,10 @@
+CREATE_USER = 'https://stellarburgers.nomoreparties.site/api/auth/register'
+LOGIN_USER = 'https://stellarburgers.nomoreparties.site/api/auth/login'
+UPDATE_USER_DATA = 'https://stellarburgers.nomoreparties.site/api/auth/user'
+CREATE_ORDER = 'https://stellarburgers.nomoreparties.site/api/orders'
+GET_ORDERS = 'https://stellarburgers.nomoreparties.site/api/orders'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,10 @@
+GET_ORDERS = 'https://stellarburgers.nomoreparties.site/api/orders'
+DELETE_USER = 'https://stellarburgers.nomoreparties.site/api/auth/user'
+
+existing_user_data = {'email': 'john_snow111@gmail.com', 'password': '111111', 'name': 'Johnny'}
+correct_ingredients = {"ingredients": ["61c0c5a71d1f82001bdaaa6d", "61c0c5a71d1f82001bdaaa70"]}
+incorrect_ingredients = {"ingredients": ["61c0c5a71d1f82001", "61c0c5af82001bdaaa70"]}
```
Урлы стоит отделить в отдельны класс от тестовых данных

AlexKitIt commented on helpers.py:
```diff
@@ -0,0 +1,40 @@
+    return random_string
+
+
+def delete_user(response_data):
+    access_token = response_data.json().get('accessToken')
+    requests.delete(DELETE_USER, headers={'Authorization': access_token})
```
Нужно исправить: логические методы необходимо вынести их helpers, они выполняют не примитивную общую логику и над ним должен быть декоратор @allure.step

AlexKitIt commented on tests/test_change_user_data.py:
```diff
@@ -0,0 +1,40 @@
+    ])
+    def test_update_data_without_auth_failed(self, update_field, new_value):
+        update_payload = {update_field: new_value}
+        response = requests.patch(UPDATE_USER_DATA, json=update_payload)
+        assert (response.status_code == 401 and
+                response.json() == {'success': False, 'message': 'You should be authorised'})
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

AlexKitIt commented on tests/test_create_user.py:
```diff
@@ -0,0 +1,48 @@
+        headers = {'Content-Type': 'application/json'}
+        response = requests.post(CREATE_USER, json=payload, headers=headers)
+        assert response.status_code == 200 and response.json().get('success') is True
+        delete_user(response)
```
Это стоит описывать в финализаторе фикстур

gestia535 committed:
Исправления в фикстурах:
убрала фикстуру для авторизации, изменила фикстуру для создания заказа
Внесены изменения в код в файлах test_change_user_data.py, test_create_order.py и test_get_user_order.py, где использовались фикстуры

gestia535 committed:
Исправления в фикстурах:
убрала фикстуру для авторизации, изменила фикстуру для создания заказа
Внесены изменения в код в файлах test_change_user_data.py, test_create_order.py и test_get_user_order.py, где использовались фикстуры
Внесены изменения в файл test_create_user.py, добавлено удаление аккаунта в код. Удалена функция из helpers.py
